plugins/output/github/public/public.go

```diff
@@ -0,0 +1,100 @@
+package public
+
+import (
+
+	"os"
+	"fmt"
+	"strings"
+	"context"
+
+	"golang.org/x/oauth2"
+	"github.com/google/go-github/v31/github"
+	
+	"github.com/parinithshekar/github-migration-cli/common/config"
+	logger "github.com/parinithshekar/github-migration-cli/wrap/logrus/v1"
+)
+// Public struct defines fields in github-public object
+type Public struct {
+	accountID string
+	accessToken string
+	kind string
+	api *github.Client
+	ctx context.Context
+}
+
+func (public * Public) setAPIClient() {
+	ctx := context.Background()
+
+	accessToken := os.Getenv(public.accessToken)
+
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: accessToken},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+
+	public.api = github.NewClient(tc)
+	public.ctx = ctx
+}
+
+// New returns a new github-public object
+func New(target config.Target) (*Public) {
+	var public *Public = new(Public)
+
+	// Check if env variables mentioned in config file exist
+	// check account ID env variable
+	_, exists := os.LookupEnv(target.AccountID)
+	if !exists {
+		logger.New().Errorf("Account ID not found")
+		os.Exit(1)
+	} else {
+		public.accountID = target.AccountID
+	}
+	// check access token env variable
+	_, exists = os.LookupEnv(target.AccessToken)
```

Comment:
I'm glad this is an environment var name.  Should prevent people from accidentally checking in their creds.